src/pages/games/[id]/edit.tsx

```diff
@@ -0,0 +1,180 @@
+  const [tags, setTags] = useState<z.infer<typeof tagDataSchema>[]>();
+  const [description, setDescription] = useState("");
+
+  const idSchema = z.string().length(24);
+
+  const userDataSchema = userSchema
+    .extend({
+      _id: idSchema,
+    })
+    .omit({ hashedPassword: true });
+
+  const { data: session } = useSession();
+  const currentUser = session?.user;
+  const [userData, setUserData] = useState<z.infer<typeof userDataSchema>>();
+
+  // useEffect(() => {
+  //   if (!session) {
+  //     router.push("/");
+  //   }
+  // }, [session]);
+
+  useEffect(() => {
+    if (currentUser) {
+      getUserData();
+    }
+  }, [currentUser]);
+
+  // useEffect(() => {
+  //   if (userData && userData.label !== "administrator") {
+  //     router.push("/");
+  //   }
+  // }, [userData]);
+
+  async function getUserData() {
+    try {
+      const response = await fetch(`/api/users/${currentUser?._id}`);
+      const data = await response.json();
+      setUserData(data);
+    } catch (error) {
+      console.error("Error getting user:", error);
+    }
+  }
+
+  const getGame = async () => {
+    try {
+      const response = await fetch(`/api/games/${gameID}`);
+      if (!response.ok) {
+        setError("Failed to fetch game");
+      }
+      const data = await response.json();
+      setGameData(data);
+      setName(data.name);
+      setThemes(data.themes);
+      setTags(data.tags);
+      setDescription(data.description);
+      setLoading(false);
+    } catch (error: any) {
+      setError(error.message);
+    }
+  };
+
+  const discardChanges = async () => {
+    router.push(`/games/${gameID}`);
+  };
+
+  const saveChanges = async () => {
+    const themeIds = themes?.map((theme) => {
+      return theme._id;
+    });
+    const tagIds = tags?.map((tag) => {
+      return tag._id;
+    });
+
+    const changes = {
+      name: name,
+      description: description,
+      themes: themeIds,
+      tags: tagIds,
+    };
+
+    await fetch(`/api/games/${gameID}`, {
+      method: "PUT",
+      body: JSON.stringify(changes),
+    });
+
+    router.push(`/games/${gameID}`);
+  };
+
+  if (gameID && loading) {
+    getGame();
+  }
+
+  if (error) {
+    return <div>{error}</div>;
+  }
+
+  if (loading) {
+    return <div>Loading...</div>;
+  }
+
+  if (!gameData) {
+    return <div>Game does not exist</div>;
+  }
+
+  return (
+    <div>
+      <div className="flex justify-center">
+        <input
+          className="mt-[126px] rounded-[20px] border border-solid border-grey bg-input-bg py-2.5 text-center font-sans text-[56px] font-semibold !outline-none"
+          type="text"
+          value={name}
+          onChange={(e) => setName(e.target.value)}
+        />
+      </div>
+      <div className="mx-auto flex w-[80vw] justify-end">
+        <DeleteGameComponent gameName={gameData.name} />
+      </div>
+      <TabsComponent
+        mode="edit"
+        description={description}
+        setDescription={setDescription}
```

Comment:
I remember telling you to do something like this, but since we are already passing in gameData, can we pass in `gameData` and `setGameData` instead and then just modify gameData.description inside TabsComponent directly? So TabsComponent itself can have a `const [description, setDescription] = useState("");`. We'll be adding many more props to TabsComponent so this would keep the code cleaner for the future. Unless I'm missing something that would prevent this from working, please lmk!